File: backend/app/features/dashboard/schemas/network_overview.py
from datetime import datetime
from typing import Literal, Optional

from pydantic import BaseModel, Field

OverviewSource = Literal["template", "llm"]


class NetworkOverviewStats(BaseModel):
    reporting_clients: int = Field(ge=0)
    live_total_mib_per_sec: float = Field(ge=0)
    peak_mib_per_sec: float = Field(ge=0)
    alerts_total: int = Field(ge=0)
    blocked_queries: int = Field(ge=0)
    enabled_policy_packs: int = Field(ge=0)
    elevated_behavior_clients: int = Field(ge=0)


class NetworkOverviewRead(BaseModel):
    generated_at: datetime
    period_minutes: int = Field(ge=1, le=24 * 60)
    bullets: list[str]
    stats: NetworkOverviewStats
    source: OverviewSource = "template"
    llm_model: Optional[str] = None
    review_mode: str = "template"

File: frontend/src/features/dashboard/types/networkOverview.ts
export interface NetworkOverviewStats {
  reporting_clients: number;
  live_total_mib_per_sec: number;
  peak_mib_per_sec: number;
  alerts_total: number;
  blocked_queries: number;
  enabled_policy_packs: number;
  elevated_behavior_clients: number;
}

export type NetworkOverviewSource = 'template' | 'llm';

export interface NetworkOverview {
  generated_at: string;
  period_minutes: number;
  bullets: string[];
  summary?: string | null;
  stats: NetworkOverviewStats;
  source: NetworkOverviewSource;
  llm_model: string | null;
  review_mode: string;
  llm_error: string | null;
}
